### PR TITLE
Make sawtooth cluster command less dependent on state file

### PR DESCRIPTION
**Pull request by ojalatodd**
Cluster command now stores only essential info in state file.
Other info is calculated or provided by the node controllers.
Docker and subprocess are the current node controller types.

Signed-off-by: Todd Ojala <todd@bitwise.io>

### CODE REVIEW

File: cli/sawtooth_cli/cluster.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def load_state(start=False):
             os.makedirs(dir_name)
         state = {
             'DesiredState': 'Stopped',
-            'Nodes': {}
+            # 'Nodes': {}  # get rid of Nodes in state file
```
Comment (agunde406) [inline cli/sawtooth_cli/cluster.py]: Remove commented out code.

File: cli/sawtooth_cli/cluster.py [hunk 2]
```diff
@@ -305,34 +295,13 @@ def do_cluster_stop(args):
...
-    if len(args.node_names) == 0 and len(node_names) == 0:
-        for node_name in state_nodes:
-            state_nodes[node_name]["Status"] = "Unknown"
-
-    # If none of the nodes are running set overall State to Stopped
-    state["DesiredState"] = "Stopped"
-    for node in state_nodes:
-        if state_nodes[node]["Status"] == "Running":
-            state["DesiredState"] = "Running"
+    state["DesiredState"] = "Stopped"  # Stop using this in state file
```
Comment (agunde406) [inline cli/sawtooth_cli/cluster.py]: Why is this still in the code if we want to "# Stop using this in state file"?

File: manage/sawtooth_manage/subproc.py [hunk 3]
```diff
@@ -115,26 +119,27 @@ def start(self, node_args):
...
-
-            pid = handle.pid
-            state['Nodes'][node_name]['pid'] += [pid]
+            subprocess.Popen((executable,) + flags)
 
         self._save_state(state)
 
         if node_args.genesis:
             time.sleep(5)
+        time.sleep(5)
```
Comment (agunde406) [inline manage/sawtooth_manage/subproc.py]: Why is this sleep needed?

File: cli/sawtooth_cli/cluster.py (hunk 1 above)
Comment (ojalatodd) [inline cli/sawtooth_cli/cluster.py]: Sorry. Thought I'd scanned for that.

File: cli/sawtooth_cli/cluster.py (hunk 2 above)
Comment (ojalatodd) [inline cli/sawtooth_cli/cluster.py]: Again, my oversight  - I decided we may want to leave DesiredState in the state file, as there is no other way to determine if a process running subprocess has died. I'll remove comment.

File: cli/sawtooth_cli/cluster.py (hunk 1 above)
Comment (ojalatodd) [inline cli/sawtooth_cli/cluster.py]: Removed

File: cli/sawtooth_cli/cluster.py (hunk 2 above)
Comment (ojalatodd) [inline cli/sawtooth_cli/cluster.py]: Removed comment.

File: manage/sawtooth_manage/subproc.py (hunk 3 above)
Comment (ojalatodd) [inline manage/sawtooth_manage/subproc.py]: Removed.

**Pull request merged**